File: backend/src/main/java/com/festago/festival/dto/FestivalsResponse.java
package com.festago.festival.dto;

import static java.util.stream.Collectors.collectingAndThen;
import static java.util.stream.Collectors.toList;

import com.festago.festival.domain.Festival;
import java.util.List;

public record FestivalsResponse(
    List<FestivalResponse> festivals) {

    public static FestivalsResponse from(List<Festival> festivals) {
        return festivals.stream()
            .map(FestivalResponse::from)
            .collect(collectingAndThen(toList(), FestivalsResponse::new));
    }
}

File: android/festago/data/src/main/java/com/festago/festago/data/dto/festival/FestivalsResponse.kt
package com.festago.festago.data.dto.festival

import com.festago.festago.domain.model.festival.FestivalsPage
import kotlinx.serialization.Serializable

@Serializable
data class FestivalsResponse(
    val last: Boolean,
    val content: List<FestivalResponse>,
) {
    fun toDomain() = FestivalsPage(
        isLastPage = last,
        festivals = content.map { it.toDomain() },
    )
}
